### PR TITLE
feat: add `UserCard` component to storybook design system

### LINKED ISSUE #1288: Feature: Highlights page user card component
### Type of feature

🍕 Feature

### Current behavior

This feature request is not related to any problems, it is part of the Highlights page redesign.

### Suggested solution

For the new Highlights page, we need to build a new User Card component displaying followers, following and highlights count, as well as the user profile, name and handle.

Mockup:
image

Empty State:
image

[Figma file](https://www.figma.com/file/OpVX6WT7dmWqnwRuEvADMF/Design-Lab?type=design&node-id=9002%3A16522&mode=design&t=nVDT2ovkFo1cL7LU-1)

This issue is a child to issue #1289 

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by OgDev-01**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description
Add UserCard component and its story to the project. UserCard is a reusable component that displays user information such as name, username, avatar, followers count, following count, and highlights count. The component uses the getAvatarByUsername function from the lib/utils/github module to get the user's avatar image. The component is designed using Tailwind CSS classes and is responsive. The story file includes two stories, one with activities and one without activities.

_Generated using [OpenSauced](https://opensauced.ai/)._

## Related Tickets & Documents
Closes #1288 

### With activities
image

### No activities
image

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [x] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Review by jpmcb — APPROVED** [review]
🕺🏼

Very nice!!

**Review by bdougie — APPROVED** [review]
Thanks for turning this around.

File: components/atoms/UserCard/user-card.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            src={avatarUrl}
+            alt={`${username}'s avatar image`}
+          />
+          <div>
+            <h3 className="text-lg text-center">{name}</h3>
+            <p className="text-center text-light-slate-9">{`@${username}`}</p>
+          </div>
+        </div>
+        <div className="flex items-center gap-5 text-lg text-center">
+          <div>
```
Comment (NsdHSO) [inline components/atoms/UserCard/user-card.tsx:32]: IMHO could you create a loop, and delete 8 lines of code

File: stories/atoms/user-card.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+const UserCardTemplate: ComponentStory<typeof UserCard> = (args) => <UserCard {...args} />;
+
+export const UserCardStory = UserCardTemplate.bind({});
+
+export const NoActivities = UserCardTemplate.bind({});
+
+UserCardStory.args = {
+  username: "foxyblocks",
+  name: "Chris Schlensker",
+  followersCount: 2,
```
Comment (NsdHSO) [inline stories/atoms/user-card.stories.tsx:19]: metaData : [{name: 'Followers', count: 2}, ....]

**Pull request merged**